lib/puppetx/puppetlabs/strings/yard/templates/default/puppetnamespace/setup.rb

```diff
@@ -1,7 +1,87 @@
 include T('default/module')
 
+require File.join(File.dirname(__FILE__),'../html_helper')
+require File.join(File.dirname(__FILE__),'../template_helper')
+
 def init
   sections :header, :box_info, :pre_docstring, T('docstring'),
     :method_summary, [:item_summary],
     :method_details_list, [T('method_details')]
+
+  @methods = object.children
+  @template_helper = TemplateHelper.new
+end
+
+def header
+  if @methods[0]['puppet_4x_function']
+    @header_text = "Puppet 4 Functions"
+  else
+    @header_text = "Puppet 3 Functions"
+  end
```

Comment:
ok, maybe just add a comment that says `# The list is expected to only contain one type of functions` to help the next reader understand that the logic is not flawed.